apps/wallet/src/ui/app/staking/getStakingRewards.ts

```diff
@@ -15,11 +15,9 @@ export function getStakingRewards(
         delegation.delegation_status === 'Pending'
     )
         return 0;
-    const validatorAddress = delegation.staked_sui.validator_address;
+    const pool_id = delegation.staked_sui.pool_id;
     const validator = activeValidators.find(
-        ({ fields }) =>
-            fields.delegation_staking_pool.fields.validator_address ===
-            validatorAddress
+        ({ fields }) => fields.delegation_staking_pool.fields.id.id === pool_id
```

Comment:
@Jibz1 does this look okay?

Reply:
Can we derive the validator meta data from the staking pool id?

Reply:
I added a Table `staking_pool_mappings` storing mappings from staking pool id to validator address so I think the answer is yes. It uses dynamic fields though.

Reply:
I see there are a few other places where validator_address is being used so I'm keeping it in the staked sui for now. We can work on removing these usages.